cmd/globaldns.go

```diff
@@ -0,0 +1,931 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/rancher/cli/cliclient"
+	managementClient "github.com/rancher/types/client/management/v3"
+	"github.com/urfave/cli"
+)
+
+const (
+	alidnsProvider     = "alidns"
+	cloudflareProvider = "cloudflare"
+	route53Provider    = "route53"
+
+	envAlibabaCloudAccessKey = "ALI_ACCESS_KEY_ID"
+	envAlibabaCloudSecretKey = "ALI_ACCESS_KEY_SECRET"
+	envAwsAccessKey          = "AWS_ACCESS_KEY_ID"
+	envAwsSecretKey          = "AWS_SECRET_ACCESS_KEY"
+	envCloudflareAPIEmail    = "CF_API_EMAIL"
+	envCloudflareAPIKey      = "CF_API_KEY"
+
+	argAwsAccessKey          = "aws-access-key"
+	argAwsSecretKey          = "aws-secret-key"
+	argCloudflareAPIEmail    = "cloudflare-api-email"
+	argCloudflareAPIKey      = "cloudflare-api-key"
+	argAlibabaCloudAccessKey = "alibabacloud-access-key-id"
+	argAlibabaCloudSecretKey = "alibabacloud-access-key-secret"
+
+	argFQDN            = "fqdn"
+	argMember          = "member"
+	argMultiClusterApp = "multi-cluster-app"
+	argProject         = "project"
+	argProvider        = "provider"
+	argRootDomain      = "root-domain"
+	argTTL             = "ttl"
+	argType            = "type"
+
+	memberAccessTypeOwner = "owner"
+)
+
+type globalDNSHolder struct {
+	ID       string
+	Provider managementClient.GlobalDNSProvider
+	Entry    managementClient.GlobalDNS
+	Target   string
+}
+
+func GlobalDNSCommand() cli.Command {
+	return cli.Command{
+		Name:  "globaldns",
+		Usage: "Operations on global DNS providers and entries",
+		Subcommands: []cli.Command{
+			{
+				Name:    "providers",
+				Aliases: []string{"provider"},
+				Usage:   "Operations on global DNS providers",
+				Action:  defaultAction(globalDNSProviderLs),
+				Subcommands: []cli.Command{
+					{
+						Name:   "ls",
+						Usage:  "List global DNS providers",
+						Action: globalDNSProviderLs,
+						Flags: []cli.Flag{
+							formatFlag,
+							quietFlag,
+						},
+					},
+					{
+						Name:      "create",
+						Usage:     "Create a global DNS provider",
+						Action:    globalDNSProviderCreate,
+						ArgsUsage: "[NAME]",
+						Flags: []cli.Flag{
+							cli.StringFlag{
+								Name:  argType,
+								Usage: "Global DNS provider type, available options are \"route53\", \"cloudflare\" and \"alidns\"",
+							},
+							cli.StringFlag{
+								Name:  argRootDomain,
+								Usage: "Set root domain of a global DNS provider",
+							},
+							cli.StringSliceFlag{
+								Name:  argMember,
+								Usage: "Set members of the global DNS provider, can be used multiple times",
+							},
+							cli.StringFlag{
```

Comment:
Put the 6 key flags together outside the command creation then add the reference like it's done with formatFlag and quietFlag.